### ISSUE TITLE
Please add search to the documentation pages

### ISSUE THREAD

**Issue by nickcollier** [google/googletest#4044]
The new docs look great and work well when reading sections top to bottom but if I am already using the library and need some help/clarification on a specific item (e.g. a specific matcher, action etc.) it would be very helpful to search all the documentation pages and see everywhere it is referenced.

**Comment by theranamuskan** [comment]
> The new docs look great and work well when reading sections top to bottom but if I am already using the library and need some help/clarification on a specific item (e.g. a specific matcher, action etc.) it would be very helpful to search all the documentation pages and see everywhere it is referenced.

Can you please add some screenshots??

**Comment by Spongeloaf** [comment]
Yes please! Searching the documentation is a basic feature that is sorely missing.

**Comment by msfrank** [comment]
hello- i also have found the (otherwise excellent) documentation hard to navigate when looking for something specific. I decided to try rendering the existing docs in an alternative way using sphinx. the results are here:

https://msfrank.github.io/better-googletest-docs/index.html

the repo where i'm generating this from is here: https://github.com/msfrank/better-googletest-docs

i have some ideas for further improving the usability (inject a local TOC at the top of each page, build out the index to easily search for symbols, maybe integrate google search). so far its already useful to me; i'd be interested to hear whether others find it useful too.

(btw, regarding the repo name, i realized after the fact that its pretty rude to the contributors and the community to call it 'better docs'. it's also inaccurate, since i'm not even changing any of the content, just some structure. i picked it without really thinking while i was busy hacking on the idea and needed a repo name for github. i'm happy to change it to something else, like 'community-googletest-docs'.)
